packages/office-addin-dev-certs/src/verify.ts

```diff
@@ -1,72 +1,83 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
-
+ 
 import { execSync } from "child_process";
 import * as crypto from "crypto";
 import * as fs from "fs";
 import * as path from "path";
 import * as defaults from "./defaults";
-
+import { uninstallExpiredCaCertificate } from './uninstall'
+ 
 function getVerifyCommand(): string {
     switch (process.platform) {
        case "win32":
           const script = path.resolve(__dirname, "..\\scripts\\verify.ps1");
           return `powershell -ExecutionPolicy Bypass -File "${script}" "${defaults.certificateName}"`;
        case "darwin": // macOS
-          return `security find-certificate -c '${defaults.certificateName}' -p | openssl x509 -checkend 86400 -noout`;
+          return `security find-certificate -c '${defaults.certificateName}' -p | openssl x509 -noout`;
        default:
           throw new Error(`Platform not supported: ${process.platform}`);
     }
 }
-
+ 
 export function isCaCertificateInstalled(): boolean {
     const command = getVerifyCommand();
-
+ 
     try {
         const output = execSync(command, {stdio : "pipe" }).toString();
         if (process.platform === "darwin") {
-            return true;
+            // Ceritificate exists. Now check and see if it's expired and remove it if is.
+            try {
+                const command = `security find-certificate -c '${defaults.certificateName}' -p | openssl x509 -checkend 86400 -noout`;
```

Comment:
can you move this into a function isCaCertificateExpired() and then have the check be:

// on Mac, remove certificate if expired
if (process.platform === darwin) {
  if (isCaCertificateExpired()) {
     uninstallExpiredCertificate();
   }
}

Reply:
My biggest concern with this is that the isCaCertificateInstalled() function is adding a side-effect which removes a cert. That really shouldn't be done inside this function as a side-effect. Is there a way to move this out of this function and call it for mac from another place?